conman/routes/admin.py

```diff
@@ -0,0 +1,53 @@
+"""
+Django admin integration for Routes.
+
+Because Route is based on django-polymorphic, their documentation
+should really help in understanding the code here:
+
+https://django-polymorphic.readthedocs.io/en/stable/admin.html
+"""
+
+from django.apps import apps
+from django.conf import settings
+from django.contrib import admin
+from polymorphic.admin import (
+    PolymorphicChildModelAdmin,
+    PolymorphicParentModelAdmin,
+)
+
+from .models import Route
+
+
+class RouteChildAdmin(PolymorphicChildModelAdmin):
+    """
+    Base admin class for all Route subclasses.
+
+    Every subclass of Route that needs to be represented in the admin would
+    benefit from using a ModelAdmin that is a subclass of this. eg:
+
+        @django.contrib.admin.register(CustomRoute)
+        class CustomRouteAdmin(RouteChildAdmin):
+            pass
+    """
+    base_fields = ('url',)
+    base_model = Route
+
+
+@admin.register(Route)
+class RouteParentAdmin(PolymorphicParentModelAdmin):
+    """ModelAdmin for the base Route model."""
+    base_model = Route
+    list_display = (
+        'url',
+        'polymorphic_ctype',
+    )
+    list_filter = (
+        ('polymorphic_ctype', admin.RelatedOnlyFieldListFilter),
+    )
+    ordering = ('url',)
+    search_fields = ('url',)
+
+    def get_child_models(self):
+        """Return every admin-registered subclass of Route."""
+        paths = settings.CONMAN_ADMIN_ROUTES
+        return [apps.get_model(model_path) for model_path in paths]
```

Comment:
I think we can do something more clever than adding yet another setting:

```python
def get_child_models(self):
    return self.base_model.__subclasses__()
```

This avoids requiring the developer to manually hook things up when there's only one correct configuration.

We might want to explicitly exclude abstract models, but that should be simple enough.

Reply:
I agree that the setting is not a nice solution.

I did consider this, but ultimately rejected it for a couple of reasons:

1. If there's a class that you don't want in the admin, you need to unregister this, and make a subclass.
2. 2nd level subclasses would not be picked up.

Reply:
The other approach that I seriously considered was to add a django App class for this app that could be overridden, but I didn't like the idea that one would _have_ to override it.

Ultimately, I went with the setting with the expectation of removing it, because it would be the easiest to remove, and requires the least code to change on the user side when it does.

I'd like to take another approach on this, but it requires careful consideration, I think.

Reply:
Another approach might be to have our own version of the `@register` decorator, but I'm not a fan of decorators with side-effects, so I parked that.

Reply:
I also thought of having our own `@register`, but decided not to suggest it as I thought the `__subclasses__()` was better. I hadn't realised that only searched one level deep. [This](https://stackoverflow.com/a/33607093) is a nice way to get all levels.

As for the possibility that someone might want to exclude some subclasses, perhaps that's the point at which a setting is appropriate? We could also have multiple options - default to including all, but allow setting a whitelist or blacklist explicitly.

Reply:
I'm not clear what the usecase would be for excluding a `Route` subclass from the admin. Maybe extra complexity should be considered YAGNI.

Reply:
> default to including all, but allow setting a whitelist or blacklist explicitly.

That's a fantastic idea. Thanks for the neat recursive subclass method too.

I've opened an issue to continue this conversation (https://github.com/meshy/django-conman/issues/122).